### ISSUE TITLE
Recursive subscript bug

### ISSUE THREAD

**Issue by startupthekid** [ReactiveKit/ReactiveKit#8]
I discovered a strange bug when I was using `ObservableCollection`. I'm modeling data for a collection view which is typically a 2D array.

My collection declaration:
```
var elements: ObservableCollection<[ObserverableCollection<[Element]>]> = ObservableCollection([])
```

When I would go and try to remove an element like so:

```
self.elements[indexPath.section].removeAtIndex(indexPath.item)
```

`subscript` in `ObservableCollection+Array.swift` (line 64) gets called recursively until the stack blows.

However, if I do this:

```
var observable = self.elements[indexPath.section]
observable.removeAtIndex(indexPath.item)
```

Everything works just fine. 

This may all be happening because I'm modeling a 2D array `Element`s incorrectly. Ideally I'd like all the changes that happen in the inner arrays to be propagated up to the outside collection.

**Comment by srdanrasic** [comment]
Hmm, I wasn't able to reproduce your issue. I did however found something potential wrong and fixed it in v1.0.5 - please try.

Are you maybe modifying the array within the `observe` closure? That should be done with caution as each modification calls `observe` again.

**Comment by startupthekid** [comment]
Yeah the `v1.0.5` update worked perfectly :+1:
